src/gui/layout/qgslayoutmapwidget.cpp

```diff
@@ -1245,6 +1246,43 @@ void QgsLayoutMapWidget::mRemoveGridPushButton_clicked()
   mMapItem->update();
 }
 
+void QgsLayoutMapWidget::mCopyGridPushButton_clicked()
+{
+  QListWidgetItem *item = mGridListWidget->currentItem();
+  if ( !item )
+  {
+    return;
+  }
+
+  QgsLayoutItemMapGrid *sourceGrid = mMapItem->grids()->grid( item->data( Qt::UserRole ).toString() );
+  int i = 0;
```

Comment:
Let's be extra safe:

```suggestion
  QgsLayoutItemMapGrid *sourceGrid = mMapItem->grids()->grid( item->data( Qt::UserRole ).toString() );
  if ( !sourceGrid )
    return;
    
  int i = 0;
```